crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -544,12 +520,10 @@ pub enum Operation {
     Jeq(NodeId, BlockId), //jump on equal
     Jmp(BlockId),         //unconditional jump
     Phi { root: NodeId, block_args: Vec<(NodeId, BlockId)> },
-
-    Call(noirc_frontend::node_interner::FuncId, Vec<NodeId>), //Call a function
+    Call(noirc_frontend::node_interner::FuncId, Vec<NodeId>, Vec<ArrayId>), //Call a function
```

Comment:
Are these array returns or arrays passed into the function? If its the former, why aren't they on result instructions like other function results, and why do we need special handling of array values?

Reply:
There are arrays returned by the function call. We need special handling because we use them during inlining to avoid unecessary copy of arrays.
Because they impact the call instruction (the result of the inlining is different with or without them) it won't be correct to have them on the result instruction. For instance if you want to do cse with the call instruction, you should not do it if 2 calls don't have the same returned arrays.